programs/drift/src/controller/orders.rs

```diff
@@ -231,6 +231,26 @@ pub fn place_perp_order(
         now
     )?;
 
+    let max_oi = market.amm.max_open_interest;
+    let oi_plus_order = match params.direction {
+        PositionDirection::Long => market
+            .amm
+            .base_asset_amount_long
+            .safe_add(order_base_asset_amount.cast()?)?
+            .unsigned_abs(),
+        PositionDirection::Short => market
+            .amm
+            .base_asset_amount_short
+            .safe_sub(order_base_asset_amount.cast()?)?
+            .unsigned_abs(),
+    };
```

Comment:
you have to compare vs user existing position

if `existing_base_asset_amount` > 0

`order_base_asset_amount`